Update Product OS doc + Clean up old getting started

Lior539 opened the pull request:
Cleaning up things after updating all the product docs in my previous PR:

- Delete old getting started docs, since each product has it's own installation page now, and are much clearer how to get started now
- Update Product OS doc page ([Issue](https://github.com/PostHog/posthog.com/issues/6411))
- Reworked the landing page for ./docs a bit. (Would love feedback on this @corywatilo )
- Moved Enable Betas to "Advanced" (since it was under "Start here", but that's been deleted now)

joethreepwood reviewed (changes requested):
Most of this looks fine on first pass to me, but I'll defer to @corywatilo as he has the biggest vision on what Docs will be. 

I've marked Request Changes though, as I think we can do much better with the Product OS page even if I set aside my feelings documented elsewhere that we should remove this. 

Updating the bullet point list for this page isn't going to be sufficient in order to make it worth keeping, and at the moment this PR doesn't update it to even include all PostHog products. 

If we're keeping this page (which has proven to be wildly unhelpful for users, as per the linked issue) then we should be building out content here to show the value and become helpful. We can do this by showing how the different elements of Product OS can work together. My suggestion is that we surface some use-cases where Product OS works fully, potentially link to some customer stories, or pull together guides about the child sections (how to use HogQL, for example). We can show examples of how to use Surveys with Feature Flags and generate Insights and send them to a data warehouse, etc.

There's an argument that this would turn the page into more of a marketing asset, but I feel that's fine as Product OS is ultimately a marketing team. We need to educate users about. If we're not taking an opportunity to do this then I feel it becomes a distraction and we'd be best removing the Product OS page and categorising the child sections differently.

andyvan-ph reviewed (commented):
- Agree with @joethreepwood on the "What is Product OS?" page. If it's going to exist, it needs to be a proper page. It's the most complained about page, and it's on the top 20 most visited docs pages as well. We don't need or want an essay, but we can do better. If we can't, we shouldn't have the page.

- I agree with removing the existing 'Start here' content given it was product analytics centric, but we should still have one – it just needs to be different in focus. Maybe we return to this at a later date?

- On a purely practical point, lots of people search for "install posthog" or similar, but this PR removes that page. If only for SEO, we should still have a page called "install posthog" that directs people to where they need to go + provides some context.

andyvan-ph commented on vercel.json:
```diff
@@ -698,6 +698,38 @@
             "destination": "/docs/getting-started/send-events"
         },
+        {
+            "source": "/docs/getting-started/start-here",
+            "destination": "/docs/getting-started/install"
```
This redirects to a page we're deleting

andyvan-ph commented:
Just posting my Slack comment here since it's relevant. TL;DR, I think Product OS should have its own install page, even if it's basically the same as product analytics. I could also be terribly wrong!

![Screenshot 2023-08-30 at 19 08 23](https://github.com/PostHog/posthog.com/assets/92976667/03808f61-6ac2-415e-a025-6d16876e73a2)

Lior539 commented:
Thanks everyone for the feedback. I've made the following changes:

- Deleted the "What is Product OS" page
- On the main docs page, change the title and subtitle to better reflect what Product OS is (also removed the hedgehog image, as it was creating a lot of dead space in between the subtitle and the products section)
- Added Installation page, which points you to the installation of the other products
- Renamed some of the headings on the left hand navbar

<img width="1512" alt="Screenshot 2023-08-31 at 11 17 54 AM" src="https://github.com/PostHog/posthog.com/assets/7090054/943942cc-bf6c-4099-b99a-d3a17e99d692">
<img width="1267" alt="Screenshot 2023-08-31 at 11 31 22 AM" src="https://github.com/PostHog/posthog.com/assets/7090054/e31ff526-0875-4d88-becf-ff35ed045709">

joethreepwood reviewed (changes requested):
I'm going to be blunt with my feedback here, but it's all well intentioned. 

This is a step backwards. As per my previous review, a simple bullet point list is not sufficient from either a design, education, or content perspective. This is voted the least helpful page in the docs, and the changes don't offer any improvements.

Worse, the content here doesn't work -- the Surveys link 404s and others are basically already mirrored on the previous Overview page in a better way.  This page as proposed offers less value than it does currently.

I don't think 'Core foundation' is a useful or clear term. It actually makes things _less_ clear. 

We had a lengthy discussion in the Sprint about [what to do](https://github.com/PostHog/company-internal/issues/1121#issuecomment-1699404543) , and these changes accomplish none of them. We're really just relisting stuff that is already linked in the nav, with no stated vision of what product OS is (except for a new Overview title), and no clear CTA for first steps. We can do much better.

Other changes, now that I've had a closer look, seem minimal and of unclear value. Considering [the time invested](https://github.com/PostHog/posthog.com/issues/6154), the outcome is disappointing in my opinion.

If we can't do better than this, my suggestion is to either:

- Leave as-is and let W&D team choose how to handle as this is more clearly within their purview and stepping on toes here isn't helping.
- We return to the idea ([broached three weeks ago](https://github.com/PostHog/posthog.com/pull/6468)) of removing the Product OS page entirely and having the Product OS button in the nav direct to `/docs`. (I'm happy to keep Product OS as a term elsewhere, however).

joethreepwood commented:
Not going to go through point by point as I think it's important to let others weigh in first and this isn't something I want to dictate solutions on over Cory/Eli. However: 

> The page that was voted least helpful is the "What is Product OS page", which has been deleted in this change. So Im not sure what you mean by returning to the idea?

Returning to the idea of deleting it entirely would involve taking 'Product OS' out of the docs altogether and making this 'Product OS' section a more generic section. Call it 'How PostHog Works', for example. 

The discussion in the sprint ended with us wanting to delete the Product OS page (which this does), and state a vision of that idea somewhere else (which this doesn't). Ultimately I think deleting one of these and creating another isn't really raising the bar. 

<img width="400" alt="Screenshot 2023-08-31 at 15 40 28" src="https://github.com/PostHog/posthog.com/assets/84011561/e45eebc5-b4c4-4135-84a5-ca97d118e4f0">
<img width="400" alt="Screenshot 2023-08-31 at 15 40 20" src="https://github.com/PostHog/posthog.com/assets/84011561/f2a5d5fa-2503-4c85-81e7-0b35f578ea8d">

> This is the current behaviour?

Yep, you're right. Apologies - got confused there comparing versions side by side. 

> I disagree. Our docs were much harder to understand, and much harder to navigate. We also sorely lacked how to get started using our other SDKs (besides JS web) and code snippets, which has now been added (as time consuming as it was).

More SDK information is always good, albeit outside this PR. However, I disagree that this version is clearer.

andyvan-ph reviewed (commented):
- Instead of 'Core foundation' maybe something like 'Platform features' or 'Product OS features'? 

RE: the Install PostHog page, it makes sense to ultimately direct people to install pages for each product, but a straight up bullet point list with little context doesn't leave a good first impression. This is currently one of the most visited docs pages we have, so it ought it to look like we've made an effort to anticipate questions users might have / add some useful context.

Off the top of my head:

- Some people who end up here will be interested in the open source version, where should they go?

- Do CDP + Data warehouse belong here right now? CDP is still in preview and doesn't make sense, afaik, unless you're already using product analytics / sending us data. Data warehouse is a private beta.

- Where should people go if they have a problem with installation? Should we point them to community questions, or elsewhere?

- These are new users, is there content / guides a new user will find useful to read once they've installed PostHog?

joethreepwood commented:
@Lior539 and I just had a chat about the feedback I put above and this line in particular. 

> Other changes, now that I've had a closer look, seem minimal and of unclear value. Considering https://github.com/PostHog/posthog.com/issues/6154, the outcome is disappointing in my opinion.

Lior and I went over it in more detail, but wanted to leave a note so it was clear I was speaking only in regard to the changes on the Product OS pages/topic, rather than the wider work in docs and around SDKs (which I haven't reviewed). Lior and I are on the same page with it now, but in the spirit of transparency it felt worth being clear that some too-fast editing made it sound more aggressive and less well-intentioned than intended! 🙏

corywatilo commented:
I’m glad this topic is being discussed because the current live version was a very rudimentary v1 back when we were launching the new site and I was trying to get the structure right for where we’re at and what’s to come.

I don’t think this is quite nailed yet, and I would agree most of the changes aren’t clear improvements. I think the best way forward might be to add some more content to the existing *What is Product OS?* page.

Here’s a brain dump of context for why it’s set up how it is today:

**Put yourself in the mind of a new user**

The first thing you expect to see when clicking the *Docs* link in the main nav is some sort of overview of everything that’s available. What I was specifically trying to *avoid* was “pulling a fast one” by landing people inside Product OS, something they weren’t looking for and have no context for what it is. Better to have a true docs index page that guides you to what you might be looking for.

**Are you actually installing “Product OS?”**

While technically you *are* “installing” Product OS by adding the code to your site, I don’t think we need to educate new users on this yet. 

Just go install the snippet - that’s the most important part since you need that to set up any product.

“Oh btw now that you’ve chosen your library and pushed your tracking code, you can use all these other super powerful platform features! We call the whole thing Product OS.”

It’s more of a follow-on concept than something you need to understand at the top.

(Microsoft didn’t sell the Windows Registry as a core feature, but it’s useful to understand later down the road.)

**Singular install instructions vs separate**

Each product *does* have its own features you can set up, so after you get the core code up and running and are sending events, that’s when it makes sense to jump to the setup instructions for a specific product. 

Though the basic setup instructions should still be referenced in each product in case that’s your entry point. But there should be clear delineation between “If you’ve never set up PostHog yet, read these instructions. But if PostHog is already installed, you can skip forward to the next step.”

---

So I think the most rational thing to do here might just be to expand upon the Product OS page to be more clear about:

- It’s like an operating system for your product usage data
- By installing the tracking code, sending events, and identifying users, you can:
    1. use this data in any product
    2. pull data in/push it out anywhere *without* actually using a product
    3. use the API to retrieve anything you’ve sent
    4. build your own apps using this data

**More historical context**

In the past I floated a more stark distinction between the Product OS platform and each individual product, trying to make it more like Windows:

- You’re buying a platform
- It comes with some apps
- You can install other apps
- And tap into any data the system has

It was covered [in this proposal](https://github.com/PostHog/product-internal/issues/474) which didn’t go anywhere since it’s a huge shift (but is worth peeking at if you haven’t seen it yet!). Ultimately if we do want to be seen as “The product platform to build on”, we need to make it more obvious that our company isn’t here to just build everything ourselves - you can build on it too. You don’t need to manage all the data - you can just build a UI that connects to our infra and ship an MVP faster. (Think: Shopify. Great core platform, and a great app ecosystem that really helped the stickiness of their platform.) (Of course, this is a larger conversation than this thread but I think @jamesefhawkins is generally on the same page. But we’re not quite there yet.)

andyvan-ph commented on contents/docs/product-os/installation.md:
```diff
@@ -0,0 +1,16 @@
+---
+title: Installation
```
```suggestion
title: Install PostHog
```

Just for consistency / SEO basically

andyvan-ph commented on vercel.json:
```diff
@@ -698,6 +698,42 @@
+        },
+        {
+            "source": "/docs/getting-started/install",
+            "destination": "/docs"
```
```suggestion
            "destination": "/docs/product-os/installation"
```

andyvan-ph replied:
This is the correct redirect for this page

andyvan-ph commented:
> 2\. The work in step 1 will take a few sprints. In the mean time, what should we do with this PR? My suggestion that we ship it as is for now (or with minor tweaks)

I think this is fine so long as we improve this installation page a little bit more before merging. It doesn't need to be radically different, it just needs a bit more TLC. Lots of people will see this page, so it's worth spending a bit more time so we leave a good first impressions.

corywatilo commented:
I still don't think this...

<img width="1226" alt="image" src="https://github.com/PostHog/posthog.com/assets/154479/5b9e37fa-c92a-4787-9b8f-bffef30c47ea">

is an improvement over this.

<img width="1219" alt="image" src="https://github.com/PostHog/posthog.com/assets/154479/0a728f25-4aaa-49d1-9607-a9831a2e2043">

Let's punt on making changes to this page until we do a docs index page rework, if we need to do that.

Lior539 commented:
Thanks everyone for all the feedback! Going to close this PR now and work with Cory on next steps
